Add docs for distributed optimizer.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29971 Add docs for distributed optimizer.**
* #29942 Polish distributed autograd docs.

Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">

Commit:
Add docs for distributed optimizer.



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)

[ghstack-poisoned]

Commit:
Update on "Add docs for distributed optimizer."



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">


[ghstack-poisoned]

rohan-varma commented on torch/distributed/optim/__init__.py:
```diff
@@ -1,6 +1,7 @@
 """
 :mod:`torch.distributed.optim` exposes DistributedOptimizer, which takes a list
 of remote parameters and runs the optimizer locally on the workers where the
```
Should "remote parameters" link to RRefs or something telling the user how to create remote parameters?

Commit:
Update on "Add docs for distributed optimizer."



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">


[ghstack-poisoned]

Commit:
Update on "Add docs for distributed optimizer."



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">


[ghstack-poisoned]

Commit:
Update on "Add docs for distributed optimizer."



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">


[ghstack-poisoned]

rohan-varma reviewed (approved):
LG, thanks!

Commit:
Update on "Add docs for distributed optimizer."



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">


[ghstack-poisoned]

Commit:
Update on "Add docs for distributed optimizer."



Differential Revision: [D18554631](https://our.internmc.facebook.com/intern/diff/D18554631/)
<img width="1330" alt="Screen Shot 2019-11-16 at 10 55 59 AM" src="https://user-images.githubusercontent.com/9958665/68997851-dc59a800-085f-11ea-9490-548314b3fde4.png">


[ghstack-poisoned]
